storage_v2/spanstore/factory.go

```diff
@@ -17,4 +18,7 @@ type Factory interface {
 
 	// CreateTraceWriter creates a spanstore.Writer.
 	CreateTraceWriter() (Writer, error)
+
+	// CreateDependencyReader creates a dependencystore.Reader.
+	CreateDependencyReader() (dependencystore.Reader, error)
```

Comment:
@yurishkuro not sure if this belongs here since we're in `package spanstore` - let me know if you have any thoughts on how we should proceed here

Reply:
I think it needs to be a different interface

Reply:
@yurishkuro Do we want to add an interface like https://github.com/jaegertracing/jaeger/blob/main/storage/dependencystore/interface.go#L1-L22 in `storage_v2` to operate on OTLP data and use the same trick to extract the v1 reader?

Reply:
no, I think we should keep that same interface for now. Dependencies are not part of OTLP, so we always need our own model, which we have in v1 interface, so I don't see a strong reason to create v2 of it.

Reply:
sounds good! i changed the storage extension to initialize v1 and v2 factories